### PR TITLE
Moves some of TestTorchMathOps to OpInfos

### PR DESCRIPTION
**Pull request by mruberry**
This PR fixes three OpInfo-related bugs and moves some functions from TestTorchMathOps to be tested using the OpInfo pattern. The bugs are:

- A skip test path in test_ops.py incorrectly formatted its string argument
- Decorating the tests in common_device_type.py was incorrectly always applying decorators to the original test, not the op-specific variant of the test. This could cause the same decorator to be applied multiple times, overriding past applications.
- make_tensor was incorrectly constructing tensors in some cases

The functions moved are:

- asin
- asinh
- sinh
- acosh
- tan
- atan
- atanh
- tanh
- log
- log10
- log1p
- log2

In a follow-up PR more or all of the remaining functions in TestTorchMathOps will be refactored as OpInfo-based tests.

### CODE REVIEW

File: test/test_unary_ufuncs.py [hunk 1]
```diff
@@ -234,6 +212,32 @@ def _fn(t):
             actual = alt(t.clone())
             self.assertEqual(actual, expected, rtol=0, atol=0)
 
+    # Helper for comparing torch tensors and numpy arrays
+    # TODO: should this or assertEqual also validate that strides are equal?
+    def assertEqualHelper(self, actual, expected, msg, *, dtype, exact_dtype=True, **kwargs):
```
Comment (mruberry) [inline test/test_unary_ufuncs.py:217]: should assertEqual handle NumPy arrays directly?

File: torch/testing/_internal/common_methods_invocations.py [hunk 2]
```diff
@@ -211,13 +214,57 @@ def sample_inputs(self, device, dtype, requires_grad=False):
...
+                                active_if=TEST_WITH_ROCM),
                        SkipInfo('TestGradients', 'test_fn_grad',
                                 dtypes=[torch.cdouble], active_if=IS_WINDOWS),
                        SkipInfo('TestGradients', 'test_method_grad',
                                 dtypes=[torch.cdouble], active_if=IS_WINDOWS),
                        SkipInfo('TestGradients', 'test_inplace_grad',
                                 dtypes=[torch.cdouble], active_if=IS_WINDOWS),
                    )),
+    # NOTE: the derivative for inplace acosh is not implemented
+    # TODO: working here
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:228]: remove this (debugging spew)

File: torch/testing/_internal/common_methods_invocations.py [hunk 3]
```diff
@@ -240,6 +287,52 @@ def sample_inputs(self, device, dtype, requires_grad=False):
...
+                       SkipInfo('TestUnaryUfuncs', 'test_reference_numerics',
+                                device_type='cpu', dtypes=[torch.cfloat, torch.cdouble],
+                                active_if=IS_WINDOWS),
+                   )),
+    UnaryUfuncInfo('log10',
+                   ref=np.log10,
+                   domain=(0, float('inf')),
+                   decorators=(precisionOverride({torch.bfloat16: 1e-2}),),
+                   skips=(
+                       SkipInfo('TestUnaryUfuncs', 'test_reference_numerics',
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:307]: Eventually need to re-enable some comparison vs. reference on "test_easy_reference_numerics" or using golden values-based testing.

Comment (mruberry) [reply torch/testing/_internal/common_methods_invocations.py:307]: Next PR: "easy_reference_numerics" where only values from -10 to 10 (intersected with domain) are tested

Comment (mruberry) [reply torch/testing/_internal/common_methods_invocations.py:307]: Duplicate test_reference_numerics for easy (small, in domain?) values vs. hard (large, + outside of domain).

**Review by ngimel — APPROVED** [review]

**Comment by mruberry** [comment]
Unlanded because this was reverted due to complex failures on a build that only runs post-land (which is why the error doesn't appear on the PR). Relevant log snippets:

```
test_reference_numerics_log2_cpu_complex128 - TestUnaryUfuncsCPU
Traceback (most recent call last):
  File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_device_type.py", line 270, in instantiated_test
    result = test_fn(self, *args)
  File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_utils.py", line 528, in wrapper
    fn(*args, **kwargs)
  File "test_unary_ufuncs.py", line 273, in test_reference_numerics
    self.assertEqualHelper(actual, expected, msg, dtype=dtype)
  File "test_unary_ufuncs.py", line 222, in assertEqualHelper
    self.assertEqual(actual.item(), expected)
  File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_utils.py", line 1164, in assertEqual
    self.assertTrue(result, msg=msg)
AssertionError: False is not true : Scalars failed to compare as equal! Comparing the imaginary part nan and 0.0 gives a difference of nan, but the allowed difference with rtol=1.3e-06 and atol=1e-05 is only 1e-05!
```

and 

```
test_reference_numerics_log2_cpu_complex64 - TestUnaryUfuncsCPU
Traceback (most recent call last):
  File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_device_type.py", line 270, in instantiated_test
    result = test_fn(self, *args)
  File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_utils.py", line 528, in wrapper
    fn(*args, **kwargs)
  File "test_unary_ufuncs.py", line 273, in test_reference_numerics
    self.assertEqualHelper(actual, expected, msg, dtype=dtype)
  File "test_unary_ufuncs.py", line 222, in assertEqualHelper
    self.assertEqual(actual.item(), expected)
  File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_utils.py", line 1164, in assertEqual
    self.assertTrue(result, msg=msg)
AssertionError: False is not true : Scalars failed to compare as equal! Comparing the imaginary part nan and 0.0 gives a difference of nan, but the allowed difference with rtol=1.3e-06 and atol=1e-05 is only 1e-05!
```

**Comment by jeffdaily** [comment]
@mruberry this PR seems critical for restoring ROCm CI to a good state.  We hope your are able to land this ASAP, assuming a clean CI report.  Thanks.

**Comment by mruberry** [comment]
> @mruberry this PR seems critical for restoring ROCm CI to a good state. We hope your are able to land this ASAP, assuming a clean CI report. Thanks.

Agreed. Unfortunately it had to be reverted previously but it should land soon. If there's an issue with it landing I'll separate the ROCm fix and land it separately.

**Pull request closed without merging**